### ISSUE TITLE
Surface.Component raises (BadMapError) expected a map, got: nil in render_component

### ISSUE THREAD

**Issue by leanersoft** [msaraiva/surface#65]
After upgrading to commit bb7d1e93be7e9624dd4550262d6507411f30db2c from previous commit, tests that use `render_component/2` started failing for components with `use Surface.Component` or `use Surface.LiveComponent`, but work if that line is changed to `use Phoenix.LiveComponent`.

```elixir
defmodule SomeComponent do
  use Surface.Component

  def render(assigns) do
    ~L"""
    <div>Hello</div>
    """
  end
end

render_component(SomeComponent, %{
  __surface__: %{groups: %{__default__: %{binding: false, size: 0}}}
})
```

Test output:

```elixir
     ** (BadMapError) expected a map, got: nil
     code: render_component(SomeComponent, %{
     stacktrace:
       (elixir 1.10.1) lib/map.ex:764: Map.split(nil, [:__default__])
       (surface 0.1.0-alpha.0) lib/surface/content_handler.ex:23: Surface.ContentHandler.init_contents/1
       test/test_project_web/live/user/some_test.exs:1: SomeComponent.render/1
       (phoenix_live_view 0.11.0) lib/phoenix_live_view/utils.ex:101: Phoenix.LiveView.Utils.to_rendered/2
       (phoenix_live_view 0.11.0) lib/phoenix_live_view/test/live_view_test.ex:393: Phoenix.LiveViewTest.__render_component__/3

```

Possibly related: https://github.com/msaraiva/surface/issues/21

**Comment by leanersoft** [comment]
Here's a test that can be executed when added to component_test.exs:

```elixir
  describe "Without LiveView" do
    @surface_attributes %{groups: %{__default__: %{binding: false, size: 0}}}
    test "render stateless component with render_component/2" do
      assert render_component(Stateless, %{
        __surface__: @surface_attributes,
        label: "My label",
        class: "myclass"
      }) =~ """
      <div class="myclass">\n  <span>My label</span>\n</div>
      """
    end
  end
```

It succeeds before commit bb7d1e93be7e9624dd4550262d6507411f30db2c and fails after it.

**Comment by leanersoft** [comment]
I used the test "render stateless component" as basis. Note that I needed to add whitespace characters to make the assert pass.

**Comment by leanersoft** [comment]
Removing `__surface__` from attributes makes it fail with the same message in earlier commits, too.

**Comment by msaraiva** [comment]
Hi @leanersoft!

`groups` has been renamed to `slots`. This should work:

```
@surface_attributes %{slots: %{__default__: %{binding: false, size: 0}}}
```

**Comment by msaraiva** [comment]
Actually, I decided to add a fallback value for `assigns.__surface__.slots` so you don't need to set it anymore when using slotless components. This should work now:

```elixir
describe "Without LiveView" do
  test "render stateless component without named slots with render_component/2" do
    assert render_component(Stateless, %{label: "My label", class: "myclass"}) =~ """
            <div class="myclass">
              <span>My label</span>
            </div>
            """
  end
end
```

**Comment by msaraiva** [comment]
Resolved in 4fb6f55f0243c7fad903ddd922a4c6667a46be02
